### PR TITLE
Initial installer commit

### PR DESCRIPTION
**Pull request by drees-bittorrent**
This is an initial commit for the installer.  It does build if you have NSIS 3.0 installed.  However, it isn't finished yet.

TODO:

- Install Web-vui
- Add support for h-value
- Better documentation

In addition, I added web-vui as a submodule and updated the icon to the utorrent icon.   It just looks better than the default present icon.

### CODE REVIEW

File: install/install.nsi [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+SetCompressor LZMA
+
+Name "${PRODUCT_NAME}"
+OutFile "${INSTALLER}"
+InstallDir "$LOCALAPPDATA\${NAME}"
+
+!define START_EXE "Update.exe"
+!define FIREWALL_EXE "BitTorrentFirewallUtility.exe"
+!define UNINSTALLER "Uninstall.exe"
+!define BENCH_URL "http://i-4101.b-${BUILD_NUMBER}.next.bench.utorrent.com/e?i=4001"
```
Comment (xercesblue) [inline install/install.nsi]: probably need a new subdomain and event number

File: install/install.nsi [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+!define MUI_ABORTWARNING
+
+;--------------------------------
+;Helper functions
+
+;Var /GLOBAL UUID
+Var /GLOBAL OSV
+Var /GLOBAL LANG
+
+!macro BenchPing event status
```
Comment (xercesblue) [inline install/install.nsi]: This is kinda neat

**Comment by drees-bittorrent** [comment]
The 4101 is actually a new event number.  I didn't catch the "next" in the
domain.  I'll ask Bommuraj about that.

Dave

File: install/install.nsi [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+	RMDir /r "${TEMP_DIR}"
+
+	!insertmacro BenchPing "install" "success"
+SectionEnd
+
+;--------------------------------
+;Uninstaller Section
+
+Section "Uninstall"
+	!insertmacro initBenchPing
```
Comment (twilmoth) [inline install/install.nsi]: Do we differentiate between install/uninstall in the initBenchPing? Do we need to?

Comment (drees-bittorrent) [inline install/install.nsi]: No, not needed.  The init part just sets up the OSV and language used in all pings.  The individual pings specify install or uninstall.

**Pull request merged**